src/sqlfluff/core/errors.py

```diff
@@ -12,6 +12,12 @@ class SQLBaseError(ValueError):
 
     def __init__(self, *args, **kwargs):
         self.ignore = kwargs.pop("ignore", False)
+        pos = kwargs.pop("pos", None)
```

Comment:
Nitpick: I think the `pop()` calls could be replaced by adding:

```
pos=None, line_no=0, line_pos=0
```

to the function parameter list.

Reply:
I think this is a hangover from python2 or maybe python3.5. Back in those days you could have `*args`, `**kwargs` **and** specified keyword arguments that weren't also in `args`. That means we broke some of the Exception code. I think you're right that it should work now though 👍